netty/src/main/java/io/grpc/transport/netty/NettyServerStream.java

```diff
@@ -89,7 +89,7 @@ protected void inboundDeliveryPaused() {
   protected void internalSendHeaders(Metadata.Headers headers) {
     writeQueue.enqueue(new SendResponseHeadersCommand(id(),
         Utils.convertServerHeaders(headers), false),
-        true);
+        !getMethodType().serverSendsOneMessage());
```

Comment:
Instead of this, could sendHeaders() have a flush argument, and this logic is done in ServerImpl.ServerCallImpl?